### ISSUE TITLE
Lag spikes when typing in code blocks

### ISSUE THREAD

**Issue by fxha** [marktext/marktext#704]
### Description

When typing in code blocks Mark Text becomes laggy.

- [x] Can you reproduce the issue? 

### Steps to reproduce

Write random stuff inside a code block.

**Expected behavior:**

No lag.

### Versions

- Mark Text: v0.13.65
- Operating system: all

**Comment by Jocs** [comment]
Is it because the file is large?

**Comment by fxha** [comment]
> Is it because the file is large?

Yes the document was larger, but typing outside code blocks was fine.

**Comment by Jocs** [comment]
>Yes the document was larger, but typing outside code blocks was fine.

Yes, code block will update after every `keyup`. Because code block need Syntax highlighting.There is there better solution?

**Comment by fxha** [comment]
That's a good question and it's a difficult to answer. Maybe our new backend will fix the performance issues in general?

**Comment by Jocs** [comment]
The new muya will improve, but it will still highlight syntax the line of code after each keyboard event. But it will try to reduce the rendering of other paragraphs, but unfortunately, in the next month, I still have no time to start muya. I will spend time improving the existing Mark Text. 😞

**Comment by Jocs** [comment]
@fxha Can you close this issue? if no more info.
